cloud/scope/powervs_cluster.go

```diff
@@ -1100,7 +1101,7 @@ func (s *PowerVSClusterScope) ReconcileVPCSubnets() (bool, error) {
 		}
 	}
 	for index, subnet := range s.IBMPowerVSCluster.Spec.VPCSubnets {
-		if subnet.Name == nil {
+		if subnet.ID == nil && subnet.Name == nil {
```

Comment:
If I understand correctly, You want to fetch the name of associated subnet id correct?

Reply:
We want to set the subnet name only is ID and name are nil. With the previous condition, in a case where we set subnet ID in spec, name will be empty and here we will be setting the name which would set different subnet name than the one passed via ID. Though `subnet.Name` will not be used in flow when subnet ID is set, thought better to set name only when appropriate.